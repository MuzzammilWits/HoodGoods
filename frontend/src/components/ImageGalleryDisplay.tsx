// src/components/ImageGalleryDisplay.tsx
import React from 'react';

interface ImageGalleryDisplayProps {
    galleryImages: string[];
}

const ImageGalleryDisplay: React.FC<ImageGalleryDisplayProps> = ({ galleryImages }) => {
    if (galleryImages.length === 0) return null;

    return (
        <div className="image-gallery">
            <h2>Recently Uploaded Images (via Supabase)</h2>
            <div className="gallery-grid">
                {galleryImages.map((url, index) => (
                    <img key={index} src={url} alt={`Uploaded ${index}`} className="gallery-image" />
                ))}
            </div>
        </div>
    );
};

export default ImageGalleryDisplay;