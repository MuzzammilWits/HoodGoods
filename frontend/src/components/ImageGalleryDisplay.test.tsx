import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/vitest'; // For extended Vitest matchers
import ImageGalleryDisplay from './ImageGalleryDisplay'; // Adjust the import path as needed

describe('ImageGalleryDisplay Component', () => {
  const mockImageUrls = [
    'https://example.com/image1.jpg',
    'https://example.com/image2.png',
    'https://example.com/image3.gif',
  ];

  it('should render null if galleryImages is empty', () => {
    const { container } = render(<ImageGalleryDisplay galleryImages={[]} />);
    // The component should return null, so the container should be empty.
    expect(container.firstChild).toBeNull();
  });

  it('should render the title when images are provided', () => {
    render(<ImageGalleryDisplay galleryImages={mockImageUrls} />);
    expect(screen.getByText('Recently Uploaded Images (via Supabase)')).toBeInTheDocument();
  });

  it('should render the correct number of images', () => {
    render(<ImageGalleryDisplay galleryImages={mockImageUrls} />);
    const images = screen.getAllByRole('img');
    expect(images).toHaveLength(mockImageUrls.length);
  });

  it('should render images with correct src and alt attributes', () => {
    render(<ImageGalleryDisplay galleryImages={mockImageUrls} />);
    const images = screen.getAllByRole('img');

    images.forEach((img, index) => {
      expect(img).toHaveAttribute('src', mockImageUrls[index]);
      expect(img).toHaveAttribute('alt', `Gallery item ${index + 1}`);
    });
  });

  it('should apply inline styles to images (basic check for existence of style attribute)', () => {
    render(<ImageGalleryDisplay galleryImages={mockImageUrls.slice(0, 1)} />); // Test with one image for simplicity
    const image = screen.getByRole('img');
    expect(image).toHaveAttribute('style');
    // More specific style checks can be brittle, but you can do them if necessary:
    // expect(image.style.width).toBe('100%');
    // expect(image.style.height).toBe('150px');
    // expect(image.style.objectFit).toBe('cover');
    // expect(image.style.borderRadius).toBe('4px');
  });

  it('should apply inline styles to the section and figure elements', () => {
    const { container } = render(<ImageGalleryDisplay galleryImages={mockImageUrls} />);
    
    // Check section style
    const sectionElement = container.querySelector('section');
    expect(sectionElement).toHaveStyle('margin: 2rem 0px');

    // Check figure style
    const figureElement = container.querySelector('figure');
    expect(figureElement).toHaveStyle('display: grid');
    expect(figureElement).toHaveStyle('grid-template-columns: repeat(auto-fill, minmax(150px, 1fr))');
    expect(figureElement).toHaveStyle('gap: 1rem');
  });


  // Snapshot tests
  it('should match snapshot when galleryImages is empty (renders null)', () => {
    const { container } = render(<ImageGalleryDisplay galleryImages={[]} />);
    expect(container.firstChild).toMatchSnapshot(); // Will be null
  });

  it('should match snapshot when images are provided', () => {
    const { container } = render(<ImageGalleryDisplay galleryImages={mockImageUrls} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

