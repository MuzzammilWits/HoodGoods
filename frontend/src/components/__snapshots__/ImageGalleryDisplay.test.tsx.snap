// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ImageGalleryDisplay Component > should match snapshot when galleryImages is empty (renders null) 1`] = `null`;

exports[`ImageGalleryDisplay Component > should match snapshot when images are provided 1`] = `
<section
  style="margin: 2rem 0px;"
>
  <h2
    style="margin-bottom: 1rem;"
  >
    Recently Uploaded Images (via Supabase)
  </h2>
  <figure
    style="display: grid; grid-template-columns: repeat(auto-fill, minmax(150px, 1fr)); gap: 1rem;"
  >
    <img
      alt="Gallery item 1"
      src="https://example.com/image1.jpg"
      style="width: 100%; height: 150px; object-fit: cover; border-radius: 4px;"
    />
    <img
      alt="Gallery item 2"
      src="https://example.com/image2.png"
      style="width: 100%; height: 150px; object-fit: cover; border-radius: 4px;"
    />
    <img
      alt="Gallery item 3"
      src="https://example.com/image3.gif"
      style="width: 100%; height: 150px; object-fit: cover; border-radius: 4px;"
    />
  </figure>
</section>
`;
